processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -1532,6 +1537,38 @@ private void applyTargetThisMapping() {
             }
         }
 
+        /**
+         * When a single target this mapping is present, and the current target type is an enum.
+         */
+        private void applyTargetThisMappingForEnumTarget() {
+            if ( !isCorrectlyDefinedTargetThisForEnumTarget() ) {
+                return;
+            }
+
+            for ( MappingReference targetThis : this.targetThisReferences ) {
+                PropertyMapping propertyMapping = new PropertyMappingBuilder().mappingContext( ctx )
+                    .sourceMethod( method )
+                    .sourcePropertyName( "test" )
```

Comment:
This seems to have no effect in this case, however this seems fishy - maybe just not set `sourcePropertyName` at all?